### ISSUE TITLE
Commands must be placed on <th> elements when using 'execute' or 'verifyRows' commands on a <table>.

### ISSUE THREAD

**Issue by mat3332** [concordion/concordion#355]
After switching from concordion version 2.1.2 to 3.1.3 I have some issues with my tests written in markdown. This is my test: 

```
      | Testheader1 | Testheader2       |
      |-------------|-------------------|
      | TestValue1  | [value][method1]  |
      | TestValue2  | [value2][method2] |
	  
[method1]: - "c:execute=#c.setC(#TEXT)"
[method2]: - "c:execute=#c.setV(#TEXT)"
```

When running this test I get exception: 

`java.lang.RuntimeException: Commands must be placed on <th> elements when using 'execute' or 'verifyRows' commands on a <table>.
at org.concordion.internal.command.executeCommand.modificationStrategies.ExecuteCommandTableModification.populateCommandCallByColumnMap (ExecuteCommandTableModification.java:26)
at org.concordion.internal.command.executeCommand.modificationStrategies.ExecuteCommandTableModification.performModification (ExecuteCommandTableModification.java:49)
at org.concordion.internal.command.executeCommand.ExecuteCommand.modifyCommandCallTree (ExecuteCommand.java:26)
at org.concordion.api.AbstractCommandDecorator.modifyCommandCallTree (AbstractCommandDecorator.java:40)
at org.concordion.api.AbstractCommandDecorator.modifyCommandCallTree (AbstractCommandDecorator.java:40)
at org.concordion.api.CommandCall.modifyTree (CommandCall.java:132)
at org.concordion.api.CommandCall.modifyTree (CommandCall.java:137)
at org.concordion.internal.DocumentParser.modifyCommandCallTree (DocumentParser.java:50)
at org.concordion.internal.DocumentParser.parse (DocumentParser.java:45)
at org.concordion.internal.XMLSpecificationReader.readSpecification (XMLSpecificationReader.java:43)
at org.concordion.Concordion.loadSpecificationFromResource (Concordion.java:104)
at org.concordion.Concordion.getSpecification (Concordion.java:71)
at org.concordion.Concordion.hasExampleCommands (Concordion.java:86)
at org.concordion.Concordion.checkValidStatus (Concordion.java:120)
at org.concordion.integration.junit4.ConcordionRunner.<init> (ConcordionRunner.java:69)
at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0 (NativeConstructorAccessorImpl.java:-2)
at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
at java.lang.reflect.Constructor.newInstance (Constructor.java:490)
at org.junit.internal.builders.AnnotatedBuilder.buildRunner (AnnotatedBuilder.java:104)
at org.junit.internal.builders.AnnotatedBuilder.runnerForClass (AnnotatedBuilder.java:86)
at org.junit.runners.model.RunnerBuilder.safeRunnerForClass (RunnerBuilder.java:70)
at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass (AllDefaultPossibilitiesBuilder.java:37)
at org.junit.runner.Computer.getRunner (Computer.java:50)
at org.junit.runner.Computer$1.runnerForClass (Computer.java:31)
at org.junit.runners.model.RunnerBuilder.safeRunnerForClass (RunnerBuilder.java:70)
at org.junit.runners.model.RunnerBuilder.runners (RunnerBuilder.java:125)
at org.junit.runners.model.RunnerBuilder.runners (RunnerBuilder.java:111)
at org.junit.runners.Suite.<init> (Suite.java:81)
at org.junit.runner.Computer$2.<init> (Computer.java:33)
at org.junit.runner.Computer.getSuite (Computer.java:28)
at org.junit.runner.Request.classes (Request.java:77)
at org.junit.runner.JUnitCore.run (JUnitCore.java:105)
at org.junit.runner.JUnitCore.runClasses (JUnitCore.java:62)
at org.junit.runner.JUnitCore.runClasses (JUnitCore.java:49)
at org.concordion.internal.runner.DefaultConcordionRunner.runJUnitClass (DefaultConcordionRunner.java:78)
at org.concordion.internal.runner.DefaultConcordionRunner.runTestClass (DefaultConcordionRunner.java:44)
at org.concordion.internal.runner.DefaultConcordionRunner.execute (DefaultConcordionRunner.java:19)
at org.concordion.internal.command.SequentialRunStrategy.call (SequentialRunStrategy.java:10)
at org.concordion.internal.command.RunCommand.execute (RunCommand.java:81)
at org.concordion.api.AbstractCommandDecorator$2.run (AbstractCommandDecorator.java:25)
at org.concordion.internal.command.LocalTextDecorator.process (LocalTextDecorator.java:26)
at org.concordion.api.AbstractCommandDecorator.execute (AbstractCommandDecorator.java:23)
at org.concordion.api.AbstractCommandDecorator$2.run (AbstractCommandDecorator.java:25)
at org.concordion.internal.command.ThrowableCatchingDecorator.process (ThrowableCatchingDecorator.java:38)
at org.concordion.api.AbstractCommandDecorator.execute (AbstractCommandDecorator.java:23)
at org.concordion.api.CommandCall.execute (CommandCall.java:72)
at org.concordion.api.CommandCallList.processSequentially (CommandCallList.java:33)
at org.concordion.internal.command.SpecificationCommand.execute (SpecificationCommand.java:36)
at org.concordion.api.CommandCall.execute (CommandCall.java:72)
at org.concordion.internal.XMLSpecification.processNode (XMLSpecification.java:60)
at org.concordion.internal.XMLSpecification.processExample (XMLSpecification.java:69)
at org.concordion.Concordion.processExample (Concordion.java:92)
at org.concordion.internal.FixtureRunner.run (FixtureRunner.java:41)
at org.concordion.integration.junit4.ConcordionRunner.invoke (ConcordionRunner.java:207)
at org.concordion.integration.junit4.ConcordionRunner$1.invoke (ConcordionRunner.java:31)
at org.concordion.integration.junit4.ConcordionFrameworkMethod.invokeExplosively (ConcordionFrameworkMethod.java:58)
at org.junit.internal.runners.statements.InvokeMethod.evaluate (InvokeMethod.java:17)
at org.junit.internal.runners.statements.RunBefores.evaluate (RunBefores.java:26)
at org.junit.runners.ParentRunner$3.evaluate (ParentRunner.java:306)
at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate (BlockJUnit4ClassRunner.java:100)
at org.junit.runners.ParentRunner.runLeaf (ParentRunner.java:366)
at org.junit.runners.BlockJUnit4ClassRunner.runChild (BlockJUnit4ClassRunner.java:103)
at org.concordion.integration.junit4.ConcordionRunner.runChild (ConcordionRunner.java:191)
at org.concordion.integration.junit4.ConcordionRunner.runChild (ConcordionRunner.java:23)
at org.junit.runners.ParentRunner$4.run (ParentRunner.java:331)
at org.junit.runners.ParentRunner$1.schedule (ParentRunner.java:79)
at org.junit.runners.ParentRunner.runChildren (ParentRunner.java:329)
at org.junit.runners.ParentRunner.access$100 (ParentRunner.java:66)
at org.junit.runners.ParentRunner$2.evaluate (ParentRunner.java:293)
at org.junit.runners.ParentRunner$3.evaluate (ParentRunner.java:306)
at org.junit.runners.ParentRunner.run (ParentRunner.java:413)
at org.concordion.integration.junit4.ConcordionRunner.run (ConcordionRunner.java:150)
at org.junit.runner.JUnitCore.run (JUnitCore.java:137)
at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs (JUnit4IdeaTestRunner.java:69)
at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute (IdeaTestRunner.java:38)
at com.intellij.rt.execution.junit.TestsRepeater.repeat (TestsRepeater.java:11)
at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs (IdeaTestRunner.java:35)
at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart (JUnitStarter.java:232)
at com.intellij.rt.junit.JUnitStarter.main (JUnitStarter.java:55)`

Please help me solve this problem.

**Comment by nigelcharman** [comment]
Hi @mat3332, thanks for reporting that issue. It's hopefully fixed in the `fix-issue-355` branch.

I'll hopefully release this in a few days. If you want it earlier or are able to try it out, you can do so by either:

1. cloning the Concordion source, running `./gradlew publishToMavenLocal` and updating your build to use this version from your maven local folder (make sure you remove it from your ~/.m2 folder afterwards), or
2. using https://jitpack.io/

**Comment by nigelcharman** [comment]
Fixed in 4.0.1
